cql3/expr/prepare_expr.cc

```diff
@@ -997,7 +997,7 @@ prepare_function_call(const expr::function_call& fc, data_dictionary::database d
         .args = std::move(parameters),
         .lwt_cache_id = fc.lwt_cache_id
     };
-    if (all_terminal && fun->is_pure() && !fun->is_aggregate()) {
+    if (all_terminal && fun->is_pure() && !fun->is_aggregate() && !fun->requires_thread()) {
```

Comment:
I guess it's ok to skip an optimization, but just a thought: It's a bit odd that if the function is particularly heavy (I am guessing functions that want to run in a thread are by average heavier), then we decide not to optimize the number of calls to it.

Reply:
It would be nice to perform the optimization, but note we don't right now, so this isn't a regression.